src/bindgen/ir/function.rs

```diff
@@ -42,6 +44,10 @@ pub struct Function {
     pub annotations: AnnotationSet,
     pub documentation: Documentation,
     pub never_return: bool,
+    // Booleans to indicate whether types are associated
+    // To be resolved after all passes
+    pub ret_assoc: Option<AssocTypesKey>,
+    pub arg_assoc: Vec<Option<AssocTypesKey>>,
```

Comment:
I would rather move these into `ret` and `args`. This means you should make `ret` and `args` a tuple

Reply:
actually, embed `arg_assoc` into `FunctionArgument` struct

Reply:
actually, embed `arg_assoc` into `FunctionArgument` struct